### ISSUE TITLE
Accessing Vue HTML Elements through vm.$$ and v-el

### ISSUE THREAD

**Issue by maedi** [vuejs/Discussion#310]
I'd like to create an input field that responds to a keyboard enter by adding a new field below and focusing the cursor on that field. However ``v-el`` doesn't appear to work on sub HTML elements, just the element that corresponds to the Vue.

**HTML**
```html
<ul id="fields" v-el="thisWorks">
  <li v-repeat="fields">
    <input v-model="text" v-el="field{{$index}}" v-on="keyup:newField(this) | key 'enter'"></input>
  </li>
</ul>
```

**JS**
```javascript

var field = function() {
  this.text = '';
}
var fields = new Vue({
  el: "#fields",
  data: {
    fields: [new field()]
  },
  methods: {
    newField: function (component) {
      // focus on new field
      var self = component;
      Vue.nextTick(function () {
        // DOESN'T WORK
        //this.$$["field" + (self.$index + 1)].focus();
      });
      // add new field
      this.$data.fields.push(new field());
    }
  }
})
```

**Error**

``Uncaught TypeError: Cannot read property 'field1' of undefined``

```fields.$$.thisWorks``` is accessible.

Happy to hear a more Vue way of doing this using components etc.

**Comment by simplesmiler** [comment]
`v-el` is a "literal directive".

> Literal directives treat their attribute value as a plain string; ... Literal directives accept mustache expressions inside their value, but these expressions will be evaluated only once on first compile and do not react to data changes.

Generally speaking, DOM operations belong to directives, so I would write a custom directive, that focuses item on `bind`: https://jsfiddle.net/simplesmiler/48jxvnzt/

**Comment by swift1** [comment]
Very nice solution, but ````| key enter```` doesn't work. Replace it with ````| key 'enter'```` and everything works great. :)

**Comment by maedi** [comment]
@simplesmiler Thank you! I'll wrap my head around the code.

**Comment by swift1** [comment]
Ye, jquery will of course lead to spaghetti code. I've extended @simplesmiler's example to give you a possible solution:

**Comment by simplesmiler** [comment]
@swift1 you are correct with the `| key 'enter'`, my cdn was pointing at 0.11.8 instead of the latest.

@maedi I would probably make `focus` a part of the viewmodel, listen to the focus event of inputs, and introduce methods `focusUp` and `focusDown`.

Something like this: https://jsfiddle.net/simplesmiler/48jxvnzt/7/
Note the directive now takes a boolean value.

**Comment by appsparkler** [comment]
i guess now it is `ref` and `this.$refs`

```
...
<h1 ref="heading">Hello World!</h1>
...

```
```
// Vue Instance
...
someMethod: function( ) {
  this.$refs.heading;    // this is the hook for the <h1> element above...
}
...
```

**Comment by tm57** [comment]
@app-sparkler correct ++
